src/coordinator/roles.spec.js

```diff
@@ -29,6 +29,200 @@ describe('Coordinator/Roles', function() {
+        let promise;
+        let request;
+        let stack;
+        let organization;
+        let role;
+        let roleStackFromServer;
+        let toCamelCase;
+
+        beforeEach(function() {
+          organization = fixture.build('contxtOrganization');
+          stack = fixture.build('contxtStack');
+          role = fixture.build('contxtRole');
+
+          expectedRoleStack = fixture.build('contxtRoleStack');
+          roleStackFromServer = fixture.build(
+            'contxtRoleStack',
+            expectedRoleStack,
+            {
+              fromServer: true
+            }
+          );
+
+          request = {
+            ...baseRequest,
+            post: this.sandbox.stub().resolves(roleStackFromServer)
+          };
+          toCamelCase = this.sandbox
+            .stub(objectUtils, 'toCamelCase')
+            .callsFake((app) => expectedRoleStack);
+
+          const roles = new Roles(baseSdk, request, expectedHost);
+          promise = roles.addStack(
+            organization.id,
+            role.id,
+            stack.id,
+            expectedRoleStack.accessType
+          );
+        });
+
+        it('posts the role stack to the server', function() {
+          expect(request.post).to.be.calledWith(
+            `${expectedHost}/stacks/${stack.id}/roles/${role.id}`,
+            {
+              access_type: expectedRoleStack.accessType
+            }
+          );
+        });
+
+        it('formats the returned role stack', function() {
+          return promise.then(() => {
+            expect(toCamelCase).to.be.calledWith(roleStackFromServer);
+          });
+        });
+
+        it('returns a fulfilled promise', function() {
+          return expect(promise).to.be.fulfilled;
+        });
+      }
+    );
+
+    context('when the role ID is not provided', function() {
+      it('throws an error', function() {
+        const organization = fixture.build('contxtOrganization')
+        const stack = fixture.build('contxtStack');
+        const roleStack = fixture.build('contxtRoleStack');
+
+        const roles = new Roles(baseSdk, baseRequest, expectedHost);
+        const promise = roles.addStack(organization.id, null, stack.id, roleStack.accessType);
+
+        return expect(promise).to.be.rejectedWith(
+          'A roleId is required for adding a stack to a role.'
+        );
+      });
+    });
+
+    context('when the stack ID is not provided', function() {
+      it('throws an error', function() {
+        const organization = fixture.build('contxtOrganization')
+        const role = fixture.build('contxtRole');
+        const roleStack = fixture.build('contxtRoleStack');
+        const roles = new Roles(baseSdk, baseRequest, expectedHost);
+        const promise = roles.addStack(organization.id, role.id, null, roleStack.accessType);
+
+        return expect(promise).to.be.rejectedWith(
+          'A stackId is required for adding a stack to a role.'
+        );
+      });
+    });
+
+  });
```

Comment:
You should add test contexts for `when the organization ID is not provided` and another one for `when the access type is not provided` to ensure it defaults.

Reply:
done

Reply:
done